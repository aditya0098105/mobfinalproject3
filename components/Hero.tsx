// components/Hero.tsx
import React from "react";
import { View, Text, StyleSheet, ImageBackground } from "react-native";
import { Colors, Spacing, Radius } from "../theme";
import { LinearGradient } from "expo-linear-gradient";
import { Feather } from "@expo/vector-icons";

export default function Hero() {
  return (
    <ImageBackground
      testID="hero-banner"
      source={require("../assets/images/banner.png")} // ✅ correct path
      style={s.wrap}
      imageStyle={s.image}
    >
      <LinearGradient
        colors={["rgba(16, 24, 40, 0.05)", "rgba(16, 24, 40, 0.65)"]}
        start={{ x: 0, y: 0 }}
        end={{ x: 0, y: 1 }}
        style={s.overlay}
      >
        <Text style={s.kicker}>CityHop</Text>
        <Text style={s.lead}>Let’s</Text>
        <Text style={s.leadAccent}>Destinations today!</Text>
        <Text style={s.subtitle}>Elevate your travel plans with curated highlights, hidden gems, and boutique stays picked for you.</Text>
        <View style={s.badges}>
          <View style={s.badge}>
            <Feather name="sun" size={14} color={Colors.accent} />
            <Text style={s.badgeText}>Weekend ready</Text>
          </View>
          <View style={s.badge}>
            <Feather name="map" size={14} color="#4ADE80" />
            <Text style={s.badgeText}>Tailored guides</Text>
          </View>
        </View>
      </LinearGradient>
    </ImageBackground>
  );
}

const s = StyleSheet.create({
  wrap: {
    height: 220,
    padding: Spacing.lg,
    borderBottomLeftRadius: Radius.xl,
    borderBottomRightRadius: Radius.xl,
    overflow: "hidden",
  },
  image: {
    borderBottomLeftRadius: Radius.xl,
    borderBottomRightRadius: Radius.xl,
  },
  overlay: {
    flex: 1,
    padding: Spacing.lg,
    justifyContent: "flex-end",
    gap: Spacing.sm,
  },
  kicker: {
    color: "#fff",
    opacity: 0.85,
    fontWeight: "700",
    fontSize: 14,
    letterSpacing: 2,
    textTransform: "uppercase",
  },
  lead: {
    color: "#F8FAFC",
    fontWeight: "800",
    fontSize: 26,
    letterSpacing: 1,
  },
  leadAccent: {
    color: "#38bdf8",
    fontWeight: "900",
    fontSize: 30,
    lineHeight: 34,
  },
  subtitle: {
    color: "#F8FAFC",
    fontSize: 14,
    lineHeight: 22,
    maxWidth: 280,
  },
  badges: {
    flexDirection: "row",
    alignItems: "center",
    gap: Spacing.sm,
    marginTop: Spacing.xs,
  },
  badge: {
    flexDirection: "row",
    alignItems: "center",
    gap: 6,
    backgroundColor: "rgba(255, 255, 255, 0.15)",
    paddingHorizontal: Spacing.sm,
    paddingVertical: Spacing.xs,
    borderRadius: 999,
  },
  badgeText: {
    color: "#fff",
    fontSize: 12,
    fontWeight: "600",
    letterSpacing: 0.3,
  },
});
