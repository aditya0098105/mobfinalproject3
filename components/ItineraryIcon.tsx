import React from "react";
import { View, StyleSheet } from "react-native";
import { Colors } from "../theme";

type Props = {
  size?: number;
  backgroundColor?: string;
  strokeColor?: string;
  accentColor?: string;
};

const BASE = 36;

export default function ItineraryIcon({
  size = 36,
  backgroundColor = "#F8FAFC",
  strokeColor = Colors.primary,
  accentColor = Colors.accent,
}: Props) {
  const scale = size / BASE;

  return (
    <View
      style={[
        styles.base,
        {
          width: size,
          height: size,
          borderRadius: 8 * scale,
          paddingHorizontal: 6 * scale,
          paddingVertical: 5 * scale,
          borderWidth: 2 * scale,
          borderColor: strokeColor,
          backgroundColor,
        },
      ]}
    >
      <View
        style={[
          styles.topStrip,
          {
            borderBottomColor: strokeColor,
            paddingBottom: 4 * scale,
            marginBottom: 6 * scale,
          },
        ]}
      >
        <View
          style={[
            styles.bindingDot,
            {
              width: 6 * scale,
              height: 6 * scale,
              borderRadius: 3 * scale,
              backgroundColor: strokeColor,
            },
          ]}
        />
        <View
          style={[
            styles.bindingDot,
            {
              width: 6 * scale,
              height: 6 * scale,
              borderRadius: 3 * scale,
              backgroundColor: strokeColor,
            },
          ]}
        />
      </View>

      {[0, 1, 2].map((item) => (
        <View
          key={item}
          style={[
            styles.row,
            {
              marginTop: item === 0 ? 0 : 4 * scale,
            },
          ]}
        >
          <View
            style={[
              styles.bullet,
              {
                width: 8 * scale,
                height: 8 * scale,
                borderRadius: 4 * scale,
                backgroundColor: item === 0 ? accentColor : strokeColor,
              },
            ]}
          />
          <View
            style={[
              styles.line,
              {
                height: 3 * scale,
                borderRadius: 2 * scale,
                backgroundColor: strokeColor,
              },
            ]}
          />
        </View>
      ))}
    </View>
  );
}

const styles = StyleSheet.create({
  base: {
    justifyContent: "flex-start",
  },
  topStrip: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    borderBottomWidth: StyleSheet.hairlineWidth,
  },
  bindingDot: {},
  row: {
    flexDirection: "row",
    alignItems: "center",
    gap: 6,
  },
  bullet: {},
  line: {
    flex: 1,
  },
});
