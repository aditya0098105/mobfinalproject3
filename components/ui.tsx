// components/ui.tsx
import React from "react";
import { Text, View, TouchableOpacity, StyleSheet, ViewStyle, TextStyle } from "react-native";
import { Colors, Radius, Spacing } from "../theme";

export function H1({ children, style }: { children: React.ReactNode; style?: TextStyle }) {
  return <Text style={[styles.h1, style]}>{children}</Text>;
}
export function H2({ children, style }: { children: React.ReactNode; style?: TextStyle }) {
  return <Text style={[styles.h2, style]}>{children}</Text>;
}
export function P({ children, style }: { children: React.ReactNode; style?: TextStyle }) {
  return <Text style={[styles.p, style]}>{children}</Text>;
}

export function Card({ children, style }: { children: React.ReactNode; style?: ViewStyle }) {
  return <View style={[styles.card, style]}>{children}</View>;
}

export function Button({
  title,
  onPress,
  variant = "primary",
  compact = false,
  right,
}: {
  title: string;
  onPress: () => void;
  variant?: "primary" | "ghost" | "outline";
  compact?: boolean;
  right?: React.ReactNode;
}) {
  const base = [styles.btn, compact && { paddingVertical: 10, paddingHorizontal: 14 }];
  const variants = {
    primary: {
      backgroundColor: Colors.primary,
      shadowColor: Colors.shadow,
      shadowOpacity: 1,
      shadowRadius: 14,
      shadowOffset: { width: 0, height: 10 },
      elevation: 3,
    },
    ghost: { backgroundColor: "transparent" as const },
    outline: { backgroundColor: "transparent" as const, borderWidth: 1, borderColor: Colors.border },
  };
  const textVariants = {
    primary: { color: "#FFFFFF" }, // light theme: white text on primary
    ghost: { color: Colors.text },
    outline: { color: Colors.text },
  } as const;

  return (
    <TouchableOpacity
      accessibilityRole="button"
      accessibilityLabel={title}
      onPress={onPress}
      activeOpacity={0.85}
      style={[...base, variants[variant]]}
    >
      <Text style={[styles.btnText, textVariants[variant]]}>{title}</Text>
      {right ? <View style={{ marginLeft: 8 }}>{right}</View> : null}
    </TouchableOpacity>
  );
}

export function Pill({ label, onPress }: { label: string; onPress: () => void }) {
  return (
    <TouchableOpacity onPress={onPress} style={styles.pill} activeOpacity={0.85}>
      <Text style={styles.pillText}>{label}</Text>
    </TouchableOpacity>
  );
}

export function SectionTitle({ children }: { children: React.ReactNode }) {
  return <Text style={styles.section}>{children}</Text>;
}

export function Divider() {
  return <View style={styles.divider} />;
}

const styles = StyleSheet.create({
  // Typography
  h1: { color: Colors.text, fontSize: 24, fontWeight: "800", letterSpacing: 0.3 },
  h2: { color: Colors.text, opacity: 0.9, fontSize: 16, fontWeight: "700" },
  p: { color: Colors.textDim, fontSize: 14, lineHeight: 20 },

  // Surfaces
  card: {
    backgroundColor: Colors.card,
    borderRadius: Radius.lg,
    borderWidth: 1,
    borderColor: Colors.border,
    padding: Spacing.md,
    shadowColor: Colors.shadow,
    shadowOpacity: 1,
    shadowRadius: 18,
    shadowOffset: { width: 0, height: 12 },
    elevation: 4,
  },

  // Buttons
  btn: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    paddingVertical: 12,
    paddingHorizontal: 18,
    borderRadius: Radius.md,
  },
  btnText: { fontSize: 15, fontWeight: "700", letterSpacing: 0.3 },

  // Pills (light)
  pill: {
    paddingVertical: 10,
    paddingHorizontal: 16,
    borderRadius: 999,
    backgroundColor: Colors.cardAlt,
    borderWidth: 1,
    borderColor: Colors.border,
  },
  pillText: { color: Colors.text, fontWeight: "600", letterSpacing: 0.2 },

  // Section header + divider
  section: {
    color: Colors.text,
    fontSize: 16,
    fontWeight: "800",
    marginTop: Spacing.lg,
    marginBottom: Spacing.sm,
    letterSpacing: 0.3,
  },
  divider: {
    height: 1,
    backgroundColor: Colors.border,
    marginVertical: Spacing.md,
  },
});
