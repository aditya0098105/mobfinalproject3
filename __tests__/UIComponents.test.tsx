import React from "react";
import { Text } from "react-native";
import { fireEvent, render } from "@testing-library/react-native";
import { Card, Divider, H1, H2, P, Pill, SectionTitle } from "../components/ui";

describe("UI primitives", () => {
  it("renders typography components", () => {
    const { getByText } = render(
      <>
        <H1>Discover</H1>
        <H2>Highlights</H2>
        <P>Hidden gems</P>
      </>
    );

    expect(getByText("Discover")).toBeTruthy();
    expect(getByText("Highlights")).toBeTruthy();
    expect(getByText("Hidden gems")).toBeTruthy();
  });

  it("wraps content inside a card", () => {
    const { getByText } = render(
      <Card>
        <Text>Inside card</Text>
      </Card>
    );

    expect(getByText("Inside card")).toBeTruthy();
  });

  it("handles pill presses", () => {
    const onPress = jest.fn();
    const { getByText } = render(<Pill label="Museums" onPress={onPress} />);

    fireEvent.press(getByText("Museums"));
    expect(onPress).toHaveBeenCalledTimes(1);
  });

  it("shows section titles and dividers", () => {
    const { getByText, toJSON } = render(
      <>
        <SectionTitle>Saved places</SectionTitle>
        <Divider />
      </>
    );

    expect(getByText("Saved places")).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
