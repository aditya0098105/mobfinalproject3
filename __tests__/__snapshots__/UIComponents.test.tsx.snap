// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI primitives shows section titles and dividers 1`] = `
[
  <Text
    style={
      [
        {
          "color": "#101828",
          "fontSize": 16,
          "fontWeight": "800",
          "letterSpacing": 0.3,
          "marginBottom": 8,
          "marginTop": 20,
        },
        undefined,
      ]
    }
  >
    Saved places
  </Text>,
  <View
    style={
      [
        {
          "backgroundColor": "#E2E8F0",
          "height": 1,
          "marginVertical": 14,
        },
        undefined,
      ]
    }
  />,
]
`;
